src/compiler/crystal/semantic/top_level_visitor.cr

```diff
@@ -1025,8 +1025,11 @@ class Crystal::TopLevelVisitor < Crystal::SemanticVisitor
 
   def check_ditto(node)
     stripped_doc = node.doc.try &.strip
-    if stripped_doc == ":ditto:" || stripped_doc == "ditto"
+    if stripped_doc == ":ditto:"
       node.doc = @last_doc
+    elsif @program.wants_doc? && stripped_doc == "ditto"
+      # TODO: remove after 0.29.0
+      raise "`ditto` is no longer supported. Use `:ditto:` instead"
```

Comment:
I will say that maybe we can use warnings here :-)

Reply:
How do you will do that @bcardiff ?

Reply:
I would add in `codegen/warnings.cr` some method to emit the warning for a `ditto`.
Call that from `top_level_visitor.cr` instead of raising.

The code in `codegen/warnings.cr` has a method to ignore the emit based on location, that is the important thing to use.
The quirk is that the top_level_visitor does not need the codegen, so maybe the `SemanticVisitor` or `TopLevelVisitor` can be reopened in the `codegen/warnings.cr` file. So all the warnings methods are located near and the ignore based on location can be defined privately.

The `ignore_warning_due_to_location` uses `@program` which is available in codegen and semantic. And I guess the `@program.warnings_exclude` will be initialized.

I think is a good exercise to stress the warnings for deprecating language features. That is my main motivation to push it in this direction.

Reply:
I am not sure how to do this, even less properly :confused: 
Wouldn't it be better/simpler to have a generic way to print a warning without using annotations instead?
For example `Warning.warn "deprecated"` in Ruby or in Python:
```py
import warnings
warnings.warn("deprecated")
```

Reply:
WDYT @bcardiff ?
I said this, because having to understand the compiler internals to print a warning is quite tedious on the user point a vue.